.github/workflows/publish-crate.yml

```
@@ -0,0 +1,49 @@
+name: Publish to crates.io
+on:
+  push:
+    tags: ['v*']  # Triggers when pushing tags starting with 'v'
+jobs:
+  publish:
+    if: github.repository == 'ruby/prism'
+    runs-on: ubuntu-latest
+    environment: release  # Optional: for enhanced security
+    permissions:
+      id-token: write     # Required for OIDC token exchange
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@f4a75cfd619ee5ce8d5b864b0d183aff3c69b55a # v2.13.1
```

I've only seen these git commit hashes used by Shopify, outside of that I've only seen version numbers, which I would prefer. So this would be:

```suggestion
        uses: step-security/harden-runner@v2
```

---

I also don't like hash pinning (for public repos), release workflows are the only exception where I admit defeat and have to say I can kind of see the point. Still, unless someone actually reviews the changes before blindly merging dependabot PRs, it doesn't really do much in my eyes.

Either way, it should be consistent about it. There are a few other actions that just pin the major version, that kind of defeats the point.